org.lflang/src/org/lflang/generator/ts/TSReactorGenerator.kt

```diff
@@ -0,0 +1,715 @@
+    private fun getTargetInitializer(state: StateVar): String {
+        return getInitializerList(state).joinToString(",")
+    }
+    private fun getTargetInitializerHelper(param: Parameter,
+                                           list: List<String>): String {
+        return if (list.size == 0) {
+            errorReporter.reportError(param, "Parameters must have a default value!")
+        } else if (list.size == 1) {
+            list[0]
+        } else {
+            list.joinToString(", ", "[", "]")
+        }
+    }
+    private fun getTargetInitializer(param: Parameter): String {
+        return getTargetInitializerHelper(param, getInitializerList(param))
+    }
+    private fun getTargetInitializer(param: Parameter, i: Instantiation): String {
+        return getTargetInitializerHelper(param, getInitializerList(param, i))
+    }
+    private fun initializeParameter(p: Parameter): String {
+        return """${p.name}: ${getTargetType(p)} = ${getTargetInitializer(p)}"""
+    }
+
+    /**
+     * Return a TS type for the specified action.
+     * If the type has not been specified, return
+     * "Present" which is the base type for Actions.
+     * @param action The action
+     * @return The TS type.
+     */
+    private fun getActionType(action: Action): String {
+        if (action.type !== null) {
+            return getTargetType(action.type)
+        } else {
+            return "Present"
+        }
+    }
+
+    /**
+     * Return a TS type for the specified port.
+     * If the type has not been specified, return
+     * "Present" which is the base type for ports.
+     * @param port The port
+     * @return The TS type.
+     */
+    private fun getPortType(port: Port): String {
+        if (port.type !== null) {
+            return getTargetType(port.type)
+        } else {
+            return "Present"
+        }
+    }
+
+    private fun generateArg(v: VarRef): String {
+        return if (v.container !== null) {
+            """__${v.container.name}_${v.variable.name}"""
+        } else {
+            """__${v.variable.name}"""
+        }
+    }
+
+    // TODO(hokeun): Split this method into smaller methods.
+    fun generateReactorFederated(reactor: Reactor, federate: FederateInstance) {
+        pr("// =============== START reactor class " + reactor.name)
+
+        for (p in reactor.preambles?: emptyList()) {
+            pr("// *********** From the preamble, verbatim:")
+            pr(p.code.toText())
+            pr("\n// *********** End of preamble.")
+        }
+
+        var reactorName = reactor.name
+        if (!reactor.typeParms.isEmpty()) {
+            reactorName +=
+                reactor.typeParms.joinToString(", ", "<", ">") { it.toText() }
+        }
+        // NOTE: type parameters that are referenced in ports or actions must extend
+        // Present in order for the program to type check.
+        if (reactor.isMain()) {
+            pr("class " + reactorName + " extends __App {")
```

Comment:
This would be a good place to use a string template:
```kotlin
            pr("class $reactorName extends __App {")
```

I tend to systematically prefer templates when the templated variable is a single identifier (you don't need braces like in `${ident.something}`)

Reply:
Done.